### ISSUE TITLE
-Wdocumentation warning for valid `img` tag when building in Xcode

### ISSUE THREAD

**Issue by amja** [llvm/llvm-project#57450]
Hello! Firstly apologies if this is not the right forum for this question. I found very little information online regarding who handles documentation parsing/warning between Clang and Xcode itself. Please close if this is not relevant.

The most direct issue I am seeing is the `HTML start tag prematurely ended, expected attribute name or '>'` warning (which I am pretty sure comes from Clang) being raised when a self-closing `img` tag is used in a header docstring comment of Objective-C code: `/// <img src="filegoeshere" alt="Alt text goes here"/>`. Since self-closing `img` tags are standard HTML, I'm not sure why this is happening.

This warning does not present when using an image tag that is *not* self closing. The problem is, I am unable to do that because this is generated code. The following is less likely to relate to Clang but I will put it anyway:

I'm building DocC documentation for a Swift project. This supports the inclusion of images via markdown syntax: `![Alt text goes here](filegoeshere)`. However, this Swift class is exposed to Objective-C so a header is built for it. When the header is generated, it converts the Markdown into HTML, thus creating this self-closing `img` tag.

I already submitted a feedback request to Apple regarding this but I would appreciate any help if this does in fact relate to Clang's documentation parsing system.

Thanks!

**Comment by dwblaikie** [comment]
If you can reproduce this with godbolt.org/open source clang, it'd help clarify whether this is a clang issue, or something Apple-specific.

**Comment by amja** [comment]
@dwblaikie reproduced this via open source clang.

Created the following file (`test.m`):
```objc
#import <Foundation/Foundation.h>

/// Represents a cat.
///
/// <img src="cat.png" alt="Illustration of a cat" />
@interface Cat: NSObject

@end

int main(int argc, char** argv) {}
```

and then ran the following command: `clang -framework Foundation test.m -Wdocumentation`.

The output was:

```
/Users/amos/Desktop/test.m:5:52: warning: HTML start tag prematurely ended, expected attribute name or '>' [-Wdocumentation]
/// <img src="cat.png" alt="Illustration of a cat" />
                                                   ^
1 warning generated.
```

**Comment by dwblaikie** [comment]
Sounds like the right place for this bug, then - not sure if it's by design, or not, as I'm not familiar with this part of clang - but hopefully someone who is can chime in here.

**Comment by amja** [comment]
Looks like this was fixed in a [recent commit](https://github.com/llvm/llvm-project/commit/0cb08093365478224f2a58cece42527fb8e1199e). Closing this issue :)
